src/jaegertracing/baggage/BaggageTest.cpp

```diff
@@ -105,7 +157,19 @@ TEST(Baggage, testRemoteRestrictionManagerFunctionality)
     ASSERT_EQ(Restriction(false, 0),
               manager.getRestriction("test-service", "abc"));
     mockAgent->addBaggageRestriction("abc", Restriction(true, 1));
-    std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+    //Wait for the Update Restrictions to occurr
+    //This is done by listening to the number of update published via the Metrics
```

Comment:
I think this brings in another component that is not needed. We can use another mechanism to wait for updates.

Reply:
Hello @isaachier, what do you suggest to listen for the updates. Why do you think that listening for the metrics is a bad idea?

Reply:
It isn't a bad approach, but if we have `InMemoryReporter` designed for this use case. Why not use that? I agree this test has race conditions, but I'd rather fix them instead of using another component.

Reply:
I dont think that the test has a race condition. it's just that some of the treatment is done asynchronously by another thread. We just need to wait to observe the result of that treatment. The reporter seemed to me as a way to observe such result.